src/modules/accounting/accounting.py

```diff
@@ -0,0 +1,222 @@
+        if report_blockstamp:
+            self.process_report(report_blockstamp)
+
+            latest_blockstamp = self._get_latest_blockstamp()
+            if not self.is_extra_data_submitted(latest_blockstamp):
+
+                slots_to_sleep = self._get_slot_delay_before_data_submit(blockstamp)
+                chain_config = self._get_chain_config(blockstamp)
+                seconds_to_sleep = slots_to_sleep * chain_config.seconds_per_slot
+                logger.info({'msg': f'Sleep for {seconds_to_sleep} before sending extra data.'})
+                sleep(seconds_to_sleep)
+
+                self._submit_extra_data(report_blockstamp)
+
+    def _submit_extra_data(self, blockstamp: BlockStamp) -> None:
+        extra_data = self.lido_validator_state_service.get_extra_data(blockstamp, self._get_chain_config(blockstamp))
+
+        tx = self.report_contract.functions.submitReportExtraDataList(extra_data.extra_data)
+
+        if not variables.ACCOUNT:
+            logger.info({'msg': 'No account provided to submit extra data. Dry mode'})
+            return
+        if self.w3.transaction.check_transaction(tx, variables.ACCOUNT.address):
+            self.w3.transaction.sign_and_send_transaction(tx, variables.GAS_LIMIT, variables.ACCOUNT)
+
+    # Consensus module: main build report method
+    @lru_cache(maxsize=1)
+    def build_report(self, blockstamp: BlockStamp) -> tuple:
+        report_data = self._calculate_report(blockstamp)
+        logger.info({'msg': 'Calculate report for accounting module.', 'value': report_data})
+        return report_data.as_tuple()
+
+    # # Consensus module: if contract got report data
+    def is_main_data_submitted(self, blockstamp: BlockStamp) -> bool:
+        processing_state = self._get_processing_state(blockstamp)
+        logger.info({'msg': 'Check if main data was submitted.', 'value': processing_state.main_data_submitted})
+        return processing_state.main_data_submitted
+
+    # Consensus module: if contract could accept any sort of report
+    def is_contract_reportable(self, blockstamp: BlockStamp) -> bool:
+        is_reportable = not self.is_main_data_submitted(blockstamp) or not self.is_extra_data_submitted(blockstamp)
+        logger.info({'msg': 'Check if contract could accept report.', 'value': is_reportable})
+        return is_reportable
+
+    @lru_cache(maxsize=1)
+    def _get_processing_state(self, blockstamp: BlockStamp) -> ProcessingState:
+        ps = named_tuple_to_dataclass(
+            self.report_contract.functions.getProcessingState().call(block_identifier=blockstamp.block_hash),
+            ProcessingState,
+        )
+        logger.info({'msg': 'Fetch processing state.', 'value': ps})
+        return ps
+
+    def is_extra_data_submitted(self, blockstamp: BlockStamp) -> bool:
+        processing_state = self._get_processing_state(blockstamp)
+        return processing_state.extra_data_items_count == processing_state.extra_data_items_submitted
```

Comment:
TODO: we should check it in a separate method after new deploy. Since the third phase will distribute rewards, and we need to send tx even if no extra data